File: backend/SindhDormitory.Domain/Entities/ApplicationHostelPreference.cs
// =============================================================================
// Domain/Entities/ApplicationHostelPreference.cs
// =============================================================================
namespace SindhDormitory.Domain.Entities;

public class ApplicationHostelPreference : BaseEntity
{
    public int PrefId          { get; set; }
    public int ApplicationId   { get; set; }
    public int HostelId        { get; set; }
    public int PreferenceOrder { get; set; }  // 1 = first choice

    // Navigation
    public Application Application { get; set; } = null!;
    public Hostel      Hostel      { get; set; } = null!;
}

File: backend-node/src/entities/ApplicationHostelPreference.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { Application } from './Application';
import { Hostel } from './Hostel';

@Entity('ApplicationHostelPreferences')
export class ApplicationHostelPreference {
  @PrimaryGeneratedColumn({ name: 'PrefId' })
  prefId!: number;

  @Column({ name: 'ApplicationId', type: 'int' })
  applicationId!: number;

  @Column({ name: 'HostelId', type: 'int' })
  hostelId!: number;

  @Column({ name: 'PreferenceOrder', type: 'int', default: 1 })
  preferenceOrder!: number;

  @CreateDateColumn({ name: 'CreatedAt' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'UpdatedAt' })
  updatedAt!: Date;

  @ManyToOne(() => Application, (a) => a.preferences, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'ApplicationId' })
  application!: Application;

  @ManyToOne(() => Hostel, (h) => h.preferences)
  @JoinColumn({ name: 'HostelId' })
  hostel!: Hostel;
}
